tron/serialize/runstate/dynamodb_state_store.py

```diff
@@ -129,28 +138,45 @@ def _get_remaining_partitions(self, items: list):
         for item in items:
             remaining_items = [
                 {"key": {"S": str(item["key"]["S"])}, "index": {"N": str(i)}}
-                for i in range(1, int(item["num_partitions"]["N"]))
+                for i in range(1, int(max(item["num_partitions"]["N"], item["num_json_val_partitions"]["N"])) + 1)
```

Comment:
also, should we gate checking the json partitions behind `read_json` in case there's any issues with the writing code and we need to turn all of the json reading off?